### PR TITLE
Add generation of virtual box image

### PR DESCRIPTION
**Pull request by cfergeau**
This pull request refactors a little bit the existing code, and then generates a virtual box image tarball in addition to the libvirt one.

### CODE REVIEW

**Comment by cfergeau** [comment]
This conflicts with the other pending pull request, I'll do the necessary rebasing once one of them is pushed

**Comment by gbraad** [comment]
@cfergeau needs a rebase, right?

File: createdisk.sh [hunk 1]
```diff
@@ -107,17 +131,30 @@ VM_PREFIX=${CRC_VM_NAME}-${random_string}
...
-# Copy the master public key
-cp id_rsa_crc $tarballDirectory/
-chmod 400 $tarballDirectory/id_rsa_crc
 
-update_json_description $1
+# VirtualBox image generation
+#
+# This must be done after the generation of libvirt image as it reuses some of
+# the content of $libvirtDestDir
+vboxDestDir="crc_vbox_$(date --iso-8601)"
```
Comment (praveenkumar) [inline createdisk.sh]: shouldn't vboxDestDir also check for the tag and the have proper naming?

```
if [ -z ${GIT_TAG} ]; then
    vboxDestDir="crc_libvirt_$(date --iso-8601)"
else
    vboxDestDir="crc_libvirt_${GIT_TAG}"
fi
```

File: createdisk.sh [hunk 2]
```diff
@@ -107,17 +131,31 @@ VM_PREFIX=${CRC_VM_NAME}-${random_string}
...
+# VirtualBox image generation
+#
+# This must be done after the generation of libvirt image as it reuses some of
+# the content of $libvirtDestDir
+vboxDestDir="crc_vbox_${destDirSuffix}"
+mkdir $vboxDestDir
+generate_vbox_directory $libvirtDestDir $vboxDestDir
 
-tar cJSf $tarballDirectory.tar.xz $tarballDirectory
+tar cJSf $vboxDestDir.tar.xz $vboxDestDir
```
Comment (praveenkumar) [inline createdisk.sh]: 👍

**Pull request merged**